modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java

```diff
@@ -777,13 +792,27 @@ private void clearErrorStoreForUnmanagedIndices(DataStream dataStream) {
      * @return the write index of this data stream before rollover was requested.
      */
     private Set<Index> maybeExecuteRollover(ClusterState state, DataStream dataStream) {
-        Index currentRunWriteIndex = dataStream.getWriteIndex();
+        Set<Index> currentRunWriteIndices = new HashSet<>();
+        currentRunWriteIndices.add(maybeExecuteRollover(state, dataStream, false));
+        if (DataStream.isFailureStoreFeatureFlagEnabled()) {
+            currentRunWriteIndices.add(maybeExecuteRollover(state, dataStream, true));
+        }
+        return currentRunWriteIndices;
+    }
+
+    @Nullable
+    private Index maybeExecuteRollover(ClusterState state, DataStream dataStream, boolean rolloverFailureStore) {
+        Index currentRunWriteIndex = rolloverFailureStore ? dataStream.unsafeGetFailureStoreWriteIndex() : dataStream.getWriteIndex();
+        if (currentRunWriteIndex == null) {
+            return null;
```

Comment:
Do we want to add this `null` to `currentRunWriteIndices` in the above function? It seems that when `maybeExecuteRollover` is called and the resulting set is added to `indicesToExcludeForRemainingRun` at https://github.com/elastic/elasticsearch/blob/7706bedfe816c2415f0da04555d27ab306fdd33d/modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java#L351 the value of `affectedIndices` at https://github.com/elastic/elasticsearch/blob/7706bedfe816c2415f0da04555d27ab306fdd33d/modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java#L412
will be off by one if `indicesToExcludeForRemainingRun` contains a null. (Unless we want to count the null?)

Reply:
Definitely not, good catch. I added a check to only add the index if it's not null.